### PR TITLE
Adapt Serverless Tier in Developer Guide

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/11125
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -20,11 +20,10 @@
...
     - [使用 TiDB Cloud (Serverless Tier) 构建 TiDB 集群](/develop/dev-guide-build-cluster-in-cloud.md)
     - [使用 TiDB 的增删改查 SQL](/develop/dev-guide-tidb-crud-sql.md)
-    - 构建简单的 CRUD 应用程序
-      - [Java](/develop/dev-guide-sample-application-java.md)
-      - [Golang](/develop/dev-guide-sample-application-golang.md)
   - 示例程序
-    - [使用 Spring Boot 构建 TiDB 应用程序](/develop/dev-guide-sample-application-spring-boot.md)
+    - [Java](/develop/dev-guide-sample-application-java.md)
+    - [Golang](/develop/dev-guide-sample-application-golang.md)
+    - [Java (Spring Boot)](/develop/dev-guide-sample-application-spring-boot.md)
```
Comment (Oreoxmt) [inline TOC.md:26]:
```suggestion
    - [Java](/develop/dev-guide-sample-application-java.md)
    - [Java (Spring Boot)](/develop/dev-guide-sample-application-spring-boot.md)
    - [Golang](/develop/dev-guide-sample-application-golang.md)
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
 
     > **Note:**
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:28]:
```suggestion
    > **注意：**
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 3]
```diff
@@ -117,55 +117,32 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
...
-    git clone https://github.com/pingcap-inc/tidb-example-java.git
-    ```
-
-2. 更改连接参数。
-
-    对于非本地默认集群、TiDB Cloud 或其他远程集群，需要更改 `plain-java-jdbc/src/main/java/com/pingcap/JDBCExample.java` 内关于 Host、Port、User、Password 的参数：
+> **注意：**
+>
+> - 此处使用 TLS 连接，是 TiDB Cloud Serverless Tier 的[强制要求](https://docs.pingcap.com/tidbcloud/secure-connections-to-serverless-tier-clusters)。
+> - 如果你在连接时遇到问题，可阅读 TiDB Cloud [Serverless Tier Clusters](https://docs.pingcap.com/tidbcloud/secure-connections-to-serverless-tier-clusters) 来获得更多信息。
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:126]:
```suggestion
> - 如果你在连接时遇到问题，可阅读 [TiDB Cloud Serverless Tier 集群安全连接](https://docs.pingcap.com/tidbcloud/secure-connections-to-serverless-tier-clusters) 来获得更多信息。
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 4]
```diff
@@ -117,55 +117,32 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
...
+```sql
++-------------------+
+| Hello TiDB Cloud! |
++-------------------+
+| Hello TiDB Cloud! |
++-------------------+
+```
 
-    输出应如[预期](https://github.com/pingcap-inc/tidb-example-java/blob/main/Expected-Output.md#plain-java-jdbc)所示
+如果你看到了这样的输出。恭喜，你已经在 TiDB Cloud 上成功地运行了自己的 SQL。
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:148]:
```suggestion
如果你的实际输出与预期输出一致，表示你已经在 TiDB Cloud 上成功地运行了 SQL 语句。
```

File: develop/dev-guide-sample-application-golang.md [hunk 5]
```diff
@@ -766,52 +766,60 @@ mysql --host 127.0.0.1 --port 4000 -u root<sql/dbinit.sql
 
 <div label="使用 GORM（推荐）" value="gorm">
 
-若你使用非本地默认集群、TiDB Cloud 或其他远程集群，更改 `gorm.go` 内 `dsn` 参数值：
+若你使用 TiDB Cloud Serverless Tier 集群，更改 `gorm.go` 内 `dsn` 参数值：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:769]: 是否需要添加使用 Dedicated Tier 集群的示例？

Comment (Icemap) [reply develop/dev-guide-sample-application-golang.md:769]: Dedicated Tier 可选择加密/非加密连接，而且还可 VPC 部署，此处无法一一赘述。

File: develop/dev-guide-sample-application-spring-boot.md [hunk 6]
```diff
@@ -260,19 +260,19 @@ spring:
...
 
 那么此处应将参数更改为：
 
 ```yaml
 spring:
   datasource:
-    url: jdbc:mysql://xxx.tidbcloud.com:4000/test
-    username: root
+    url: jdbc:mysql://xxx.tidbcloud.com:4000/test?sslMode=VERIFY_IDENTITY&enabledTLSProtocols=TLSv1.2,TLSv1.3
+    username: 2aEp24QWEDLqRFs. root
```
Comment (Mini256) [inline develop/dev-guide-sample-application-spring-boot.md:275]:
```suggestion
    username: 2aEp24QWEDLqRFs.root
```

**Review by Mini256 — APPROVED** [review]
rest LGTM

File: develop/dev-guide-sample-application-golang.md [hunk 7]
```diff
@@ -766,52 +766,60 @@ mysql --host 127.0.0.1 --port 4000 -u root<sql/dbinit.sql
...
 
 {{< copyable "" >}}
 
 ```go
 dsn := "root:@tcp(127.0.0.1:4000)/test?charset=utf8mb4"
 ```
 
-若你设定的密码为 `123456`，而且从 TiDB Cloud 得到的连接字符串为：
+若你设定的密码为 `123456`，而且从 TiDB Cloud Serverless Tier 集群面板中得到的连接信息为：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:777]: 在 Cloud 集群面板中并不能直接得到下面的连接信息

File: develop/dev-guide-sample-application-golang.md [hunk 8]
```diff
@@ -766,52 +766,60 @@ mysql --host 127.0.0.1 --port 4000 -u root<sql/dbinit.sql
...
 
-```
-mysql --connect-timeout 15 -u root -h xxx.tidbcloud.com -P 4000 -p
-```
+- Endpoint: `xxx.tidbcloud.com`
+- Port: `4000`
+- User: `2aEp24QWEDLqRFs.root`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:781]:
````suggestion
```
mysql --connect-timeout 15 -u '2aEp24QWEDLqRFs.root' -h xxx.tidbcloud.com -P 4000 -D test --ssl-mode=VERIFY_IDENTITY --ssl-ca=/etc/ssl/cert.pem -p
```
````

File: develop/dev-guide-sample-application-golang.md (hunk 7 above)
Comment (Icemap) [reply develop/dev-guide-sample-application-golang.md:777]: 有的呀
image

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: develop/dev-guide-sample-application-golang.md [hunk 9]
```diff
@@ -766,52 +766,60 @@ mysql --host 127.0.0.1 --port 4000 -u root<sql/dbinit.sql
...
 
-那么此处应将参数更改为：
-
-{{< copyable "" >}}
+那么此处应将代码块更改为：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:783]:
```suggestion
那么此处应将 `mysql.RegisterTLSConfig` 和 `dsn` 更改为：
```

File: develop/dev-guide-sample-application-golang.md [hunk 10]
```diff
@@ -766,52 +766,60 @@ mysql --host 127.0.0.1 --port 4000 -u root<sql/dbinit.sql
...
 
-```
-mysql --connect-timeout 15 -u root -h xxx.tidbcloud.com -P 4000 -p
-```
+- Endpoint: `xxx.tidbcloud.com`
+- Port: `4000`
+- User: `2aEp24QWEDLqRFs.root`
 
-那么此处应将参数更改为：
+那么此处应将代码块更改为：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:812]:
```suggestion
那么此处应将 `mysql.RegisterTLSConfig` 和 `dsn` 更改为：
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 11]
```diff
@@ -117,55 +117,32 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
...
-    ```shell
-    git clone https://github.com/pingcap-inc/tidb-example-java.git
-    ```
-
-2. 更改连接参数。
-
-    对于非本地默认集群、TiDB Cloud 或其他远程集群，需要更改 `plain-java-jdbc/src/main/java/com/pingcap/JDBCExample.java` 内关于 Host、Port、User、Password 的参数：
+> **注意：**
+>
+> - 此处使用 TLS 连接，是 TiDB Cloud Serverless Tier 的[强制要求](https://docs.pingcap.com/tidbcloud/secure-connections-to-serverless-tier-clusters)。
```
Comment (ran-huang) [inline develop/dev-guide-build-cluster-in-cloud.md:125]:
```suggestion
> - 在连接到 TiDB Cloud Serverless Tier 时，[必须使用 TLS 连接](https://docs.pingcap.com/tidbcloud/secure-connections-to-serverless-tier-clusters)。
```

**Review by ran-huang — APPROVED** [review]
rest LGTM

File: develop/dev-guide-sample-application-java.md [hunk 12]
```diff
@@ -1665,9 +1665,11 @@ mysql --connect-timeout 15 -u root -h xxx.tidbcloud.com -P 4000 -p
 
 若你设定的密码为 `123456`，而且从 TiDB Cloud 得到的连接字符串为：
```
Comment (ran-huang) [inline develop/dev-guide-sample-application-java.md:1667]:
```suggestion
```

Comment (ran-huang) [inline develop/dev-guide-sample-application-java.md:1667]:
```suggestion
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 13]
```diff
@@ -16,18 +16,18 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
 
 1. 如果你还未拥有 TiDB Cloud 帐号，请先在此[注册](https://tidbcloud.com/free-trial)。
 2. 使用你的 TiDB Cloud 帐号[登录](https://tidbcloud.com/)。
-3. 在[方案](https://tidbcloud.com/console/plans)内选择免费的 **Serverless Tier** 方案，或在 [Clusters](https://tidbcloud.com/console/clusters) 页面中点击 **Create Cluster** 按钮。
-4. 在 **Create Cluster** 页面设置集群名称、云服务商（Serverless Tier 默认为 AWS）、可用区（建议就近选择）后，点击 **Create** 按钮创建 Serverless Tier 免费集群。
-5. 在 **Security Settings** 对话框中，设置密码，并添加允许连接你的集群的 IP 地址，完成后点击 **Apply**。
+3. 在 [Clusters](https://tidbcloud.com/console/clusters) 页面中点击 **Create Cluster** 按钮。
+4. 在 **Create Cluster** 页面默认选择的 **Serverless Tier**。你可以根据需要修改集群名称并选择可用区。
```
Comment (ran-huang) [inline develop/dev-guide-build-cluster-in-cloud.md:20]:
```suggestion
4. **Create Cluster** 页面默认选择 **Serverless Tier**。你可以根据需要修改集群名称并选择可用区。
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 14]
```diff
@@ -16,18 +16,18 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
...
+5. 点击 **Create** 创建免费的 Serverless Tier 集群。
 
     你的 TiDB Cloud 集群将于 30 秒后创建完毕。
 
-6. 创建完毕后，点击右上角的 **Connect** 按钮。或点击集群名称，打开集群的详情页，再点击右上角的 **Connect** 按钮。这将显示一个连接对话框。
-7. 复制连接对话框 **Step 2: Connect with a SQL client** 中的连接字符串，供后续步骤使用。
+6. 集群创建完毕后，点击 **Security Settings** 按钮，在对话框中设置连接集群的 root 密码，完成后点击 **Apply**。如果没有设置 root 密码，你将无法连接集群。
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:25]:
```suggestion
6. 集群创建完毕后，点击 **Security Settings** 按钮，在对话框中设置连接集群的 root 密码，完成后点击 **Submit**。如果没有设置 root 密码，你将无法连接集群。
```

Comment (ran-huang) [inline develop/dev-guide-build-cluster-in-cloud.md:25]:
```suggestion
6. 集群创建完毕后，点击 **Security Settings** 按钮，在对话框中设置连接集群的 root 密码，完成后点击 **Submit**。如果没有设置 root 密码，你将无法连接集群。
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**